ai-service/src/ai_service/embeddings/tests/test_encoding.py

```
@@ -0,0 +1,130 @@
+from ai_service import errors
+from ai_service.embeddings import encoding
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+class TestInputValidation:
+    """Test input validation and error handling."""
+
+    @pytest.mark.parametrize("invalid_input", ["", "   ", "\t", "\n"])
+    def test_rejects_empty_strings(self, invalid_input: str):
```

Ruff (0.12.2)

Missing return type annotation for public function `test_rejects_empty_strings`

Add return type annotation: `None`

(ANN201)